perfkitbenchmarker/linux_benchmarks/mnist_benchmark.py

```diff
@@ -0,0 +1,155 @@
+    gcloud_scopes: https://www.googleapis.com/auth/cloud-platform
+"""
+
+flags.DEFINE_string('mnist_train_file',
+                    'gs://tfrc-test-bucket/mnist-records/train.tfrecords',
+                    'mnist train file for tensorflow')
+flags.DEFINE_bool('mnist_use_tpu', True, 'Use TPUs rather than plain CPUs')
+
+
+def GetConfig(user_config):
+  """Load and return benchmark config.
+
+  Args:
+    user_config: user supplied configuration (flags and config file)
+
+  Returns:
+    loaded benchmark configuration
+  """
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def _UpdateBenchmarkSpecWithFlags(benchmark_spec):
+  """Update the benchmark_spec with supplied command line flags.
+
+  Args:
+    benchmark_spec: benchmark specification to update
+  """
+  benchmark_spec.train_file = FLAGS.mnist_train_file
+  benchmark_spec.use_tpu = FLAGS.mnist_use_tpu
+
+
+def Prepare(benchmark_spec):
+  """Install and set up MNIST on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification
+  """
+  benchmark_spec.always_call_cleanup = True
+  _UpdateBenchmarkSpecWithFlags(benchmark_spec)
+  vms = benchmark_spec.vms
+  master_vm = vms[0]
+  master_vm.RemoteCommand(
+      'git clone https://github.com/tensorflow/tpu-demos.git', should_log=True)
+  benchmark_spec.tpu = gcp_tpu.GcpCloudTpu(FLAGS.run_uri, FLAGS.project)
+  benchmark_spec.tpu.Create()
+
+
+def _CreateMetadataDict(benchmark_spec):
+  """Create metadata dict to be used in run results.
+
+  Args:
+    benchmark_spec: benchmark spec
+
+  Returns:
+    metadata dict
+  """
+  metadata = dict()
+  metadata['train_file'] = benchmark_spec.train_file
+  metadata['use_tpu'] = benchmark_spec.use_tpu
+  return metadata
+
+
+def _ExtractThroughput(output):
+  """Extract throughput from MNIST output.
+
+  Args:
+    output: MNIST output
+
+  Returns:
+    throuput float
+  """
+  regex = r'INFO:tensorflow:global_step/sec: (\S+)'
+  match = re.findall(regex, str(output))
+  return sum(float(step) for step in match) / len(match)
+
+
+def _MakeSamplesFromOutput(benchmark_spec, output):
+  """Create a sample continaing the measured MNIST throughput.
+
+  Args:
+    benchmark_spec: benchmark spec
+    output: MNIST output
+
+  Returns:
+    a Sample containing the MNIST throughput
+  """
+  metadata = _CreateMetadataDict(benchmark_spec)
+  global_step_sec = _ExtractThroughput(output)
+  return [sample.Sample('tensorflow', global_step_sec,
+                        'global_step/sec', metadata)]
+
+
+def Run(benchmark_spec):
+  """Run MNIST on the cluster.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  _UpdateBenchmarkSpecWithFlags(benchmark_spec)
+  vms = benchmark_spec.vms
+  master_vm = vms[0]
+  mnist_benchmark_dir = 'tpu-demos/cloud_tpu/models/mnist'
+  tpu_master = 'grpc://{}:8470'.format(benchmark_spec.tpu.tpu_ip)
```

Comment:
does this require grpc to be installed on the client vm?

Reply:
I didn't install grpc. I think it default installed

Reply:
Ok